src/main/java/data/Account.java

```diff
@@ -115,20 +115,16 @@ public void setAggregationMode(String aggregationMode) {
     this.aggregationMode = aggregationMode;
   }
 
-  /** Return a String representation of Account fields. */
-  public String toString() {
-    StringBuilder accountBuilder = new StringBuilder();
-    accountBuilder.append("{Legacy_Account_ID:" + getLegacyAccountID());
-    accountBuilder.append(",Next_Gen_Customer_ID:" +  getNextGenAccountID());
-    accountBuilder.append(",Settlement_Attributes:{");
-    accountBuilder.append("Currency_Code:" + getCurrency());
-    accountBuilder.append(",Direction:" + getDirection());
-    accountBuilder.append(",Entity:" + getEntity() + "}");
-    accountBuilder.append(",Settlement_Config:{");
-    accountBuilder.append("Matching_Mode:" + getMatchingMode() + "}");
-    accountBuilder.append(",Account_ID:" + getAccountID());
-    accountBuilder.append(",Aggregation_Mode:" + getAggregationMode() + "}}");
-
-    return accountBuilder.toString();
+  /** Return a JSON String representation of the fields. */
+  public String toJson() {
+    String accountJson = String.format("{\"legacy_account_id\":%s," +
```

Comment:
This looks much nicer. As a nit: you can just return String.format(....); instead of putting it into a local variable.

Reply:
(If you change it, also change vendor.java)

Reply:
Will do.